ql/src/test/org/apache/hadoop/hive/ql/exec/TestFileSinkOperator.java

```diff
@@ -183,37 +183,42 @@ public void testNonAcidRemoveDuplicate() throws Exception {
 
     JobConf jobConf2 = new JobConf(jobConf);
     jobConf2.set("mapred.task.id", "000000_1");
-    FileSinkOperator speculative = (FileSinkOperator)OperatorFactory.get(
+    FileSinkOperator op2 = (FileSinkOperator)OperatorFactory.get(
         new CompilationOpContext(), FileSinkDesc.class);
-    speculative.setConf(desc);
-    speculative.initialize(jobConf2, new ObjectInspector[]{inspector});
+    op2.setConf(desc);
+    op2.initialize(jobConf2, new ObjectInspector[]{inspector});
 
     for (Object r : rows) {
       op1.process(r, 0);
-      speculative.process(r, 0);
+      op2.process(r, 0);
     }
 
     op1.close(false);
-    // speculative task also ends successfully
-    speculative.close(false);
+    // Assume op2 also ends successfully, this happens in different containers
+    op2.close(false);
     Path[] paths = findFilesInBasePath();
     List<Path> mondays = Arrays.stream(paths)
         .filter(path -> path.getParent().toString().endsWith("partval=Monday/HIVE_UNION_SUBDIR_0"))
         .collect(Collectors.toList());
-    Assert.assertTrue(mondays.size() == 2);
+    Assert.assertEquals("Two result files were created", 2, mondays.size());
```

Comment:
"Two result files were created" -> makes me think it's unexpected, instead we expect 2 files according to the parameters, can we have rather "2 results files are expected"